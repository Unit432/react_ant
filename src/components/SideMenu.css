.trigger {
  font-size: 18px;
  line-height: 64px;
  padding: 0 24px;
  cursor: pointer;
  transition: color 0.3s;
}

.trigger:hover {
  color: #1890ff;
}

.logo {
  height: 32px;
  margin: 16px;
}

.topIconSpan {
  font-size: 27px;
  cursor: pointer;
  padding: 0 12px;
  display: inline-block;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  height: 100%;
}

.topIconSpan:hover {
  background: #e6f7ff;
}

.topIcon {
  font-size: 16px;
  vertical-align: middle;
}
