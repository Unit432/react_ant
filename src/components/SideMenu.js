import React from 'react'
import { Layout, Menu, Icon } from 'antd'
import './SideMenu.css'
import { Link } from 'react-router-dom'

const { Sider } = Layout

class SideMenu extends React.Component {
  render () {
    return (
      <div>
        <Sider trigger={null} collapsible collapsed={this.props.sideMenuState}>
          <div className='logo' />
          <Menu theme='dark' mode='inline' defaultSelectedKeys={['1']}>
            <Menu.Item key='1'>
              <Icon type='sync' />
              <span><Link to='/'>Downloads</Link></span>
            </Menu.Item>
            <Menu.Item key='2'>
              <Icon type='setting' />
              <span><Link to='/settings'>Settings</Link></span>
            </Menu.Item>
          </Menu>
        </Sider>
      </div>
    )
  }
}

export default SideMenu
