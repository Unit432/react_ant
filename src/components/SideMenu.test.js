import React from 'react'
import SideMenu from './SideMenu'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

describe("SideMenu", () => {
  it('SideMenu renders a Downloads and a Settings menu item', () => {
    const rendered_menu = renderer
      .create(<MemoryRouter><SideMenu /></MemoryRouter>)
      .toJSON();
    expect(rendered_menu).toMatchSnapshot()
  })
})
