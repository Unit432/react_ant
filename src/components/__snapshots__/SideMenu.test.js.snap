// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu SideMenu renders a Downloads and a Settings menu item 1`] = `
<div
  className="ant-layout-sider"
  style={
    Object {
      "flex": "0 0 200px",
      "maxWidth": "200px",
      "minWidth": "200px",
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="logo"
    />
    <ul
      className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
      role="menu"
      style={Object {}}
    >
      <li
        aria-disabled={undefined}
        className="ant-menu-item ant-menu-item-selected"
        onBlur={undefined}
        onClick={[Function]}
        onContextMenu={undefined}
        onFocus={undefined}
        onMouseDown={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={undefined}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        title={undefined}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="anticon anticon-sync"
          />
          <span>
            Downloads
          </span>
        </a>
      </li>
      <li
        aria-disabled={undefined}
        className="ant-menu-item"
        onBlur={undefined}
        onClick={[Function]}
        onContextMenu={undefined}
        onFocus={undefined}
        onMouseDown={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={undefined}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        title={undefined}
      >
        <a
          href="/settings"
          onClick={[Function]}
        >
          <i
            className="anticon anticon-setting"
          />
          <span>
            Settings
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
